res/config/game_script/timetable_gui.lua

```diff
@@ -1,1269 +1,1196 @@
+		debounceMin:setValue(condition[1],false)
+	end
+
+	local debounceSec = api.gui.comp.DoubleSpinBox.new()
+	debounceSec:setMinimum(0,false)
+	debounceSec:setMaximum(59,false)
+
+	debounceSec:onChange(function(value)
+		timetable.updateDebounce(lineID, stationID, 2, value, debounceType)
+		timetableChanged = true
+		timetableGUI.initStationTable()
+		timetableGUI.fillStationTable(UIState.currentlySelectedLineTableIndex, false)
+		updateAutoDebounce()
+	end)
+
+	if condition and condition[2] then
+		debounceSec:setValue(condition[2],false)
+	end
+
+	local unbunchTimeHeader = api.gui.comp.TextView.new(UIStrings.unbunch_time .. ":")
+	local textColon = api.gui.comp.TextView.new(":")
+	debounceHeader = unbunchTimeHeader
+	if debounceType == "auto_debounce" then debounceHeader = api.gui.comp.TextView.new("Margin Time:") end
+	debounceTable:addRow({debounceHeader, debounceMin, textColon, debounceSec})
+
+	if debounceType == "auto_debounce" then
+		textColon = api.gui.comp.TextView.new(":")
+		autoDebounceMin = api.gui.comp.TextView.new("--")
+		autoDebounceSec = api.gui.comp.TextView.new("--")
+		updateAutoDebounce()
+		debounceTable:addRow({unbunchTimeHeader, autoDebounceMin, textColon, autoDebounceSec})
+	end
+
+	menu.constraintHeaderTable:addRow({debounceTable})
+end
+
+-------------------------------------------------------------
+--------------------- OTHER ---------------------------------
+-------------------------------------------------------------
+function timetableGUI.popUpMessage(message, onOK)
+	debugPrint("popUpMessage")
+	if menu.popUp then
+		menu.popUp:close()
+	end
+
+	local textOK = api.gui.comp.TextView.new("OK")
+	local okButton = api.gui.comp.Button.new(textOK, true)
+	menu.popUp = api.gui.comp.Window.new(message, okButton)
+	local position = api.gui.util.getMouseScreenPos()
+	menu.popUp:setPosition(position.x, position.y)
+	menu.popUp:addHideOnCloseHandler()
+
+	menu.popUp:onClose(function()
+		onOK()
+	end)
+
+	okButton:onClick(function()
+		menu.popUp:close()
+	end)
+end
+
+function timetableGUI.popUpYesNo(title, onYes, onNo)
+	if menu.popUp then
+		menu.popUp:close()
+	end
+	
+	local popUpTable = api.gui.comp.Table.new(2, 'NONE')
+	local textYes = api.gui.comp.TextView.new("Yes")
+	local yesButton = api.gui.comp.Button.new(textYes, true)
+	local textNo = api.gui.comp.TextView.new("No")
+	local noButton = api.gui.comp.Button.new(textNo, true)
+	popUpTable:addRow({yesButton, noButton})
+
+	menu.popUp = api.gui.comp.Window.new(title, popUpTable)
+	local position = api.gui.util.getMouseScreenPos()
+	menu.popUp:setPosition(position.x, position.y)
+	menu.popUp:addHideOnCloseHandler()
+
+	local yesPressed = false
+	menu.popUp:onClose(function()
+		if yesPressed then
+			onYes()
+		else
+			onNo()
+		end
+		menu.popUp = nil
+	end)
+
+	yesButton:onClick(function()
+		yesPressed = true
+		menu.popUp:close()
+	end)
+	noButton:onClick(function()
+		menu.popUp:close()
+	end)
+end
+
+function data()
+	return {
+		guiUpdate = function()

```

Comment:
Question for my understanding:
Can `guiUpdate` be in a separate file to the `update`, `load`, and 'save' functions?
Will all those functions still get called in the same way?

Reply:
Yes. We discussed this in the TPF forum. It is better for the stability of the script if GUI elements and other function calls are created separately. The game itself creates two separate threads for this in the background anyway.